### DISCUSSION TITLE
[Feature Request] - Have an option to synchronize start time cyclus of mutiple over_switch thermostats

### DISCUSSION THREAD

**Discussion by alibaba56734** [jmcollin78/versatile_thermostat#568] [Ideas]
I have multiple rooms with radiators connected to a central boiler. Every room is controlled by a VTherm that controls the radiators of the room. When a radiator/underlying asks for heat, the centrail boiler turns on. 

I have configured the Vtherms with the over_switch configuration, so every Vtherm have its own cyclus of turning on and off. This can cause the central boiler to be turned on and off a lot.

I was wondering if it would be possible to synchronize the start time of the Vtherm over_switch cyclus, so the times of turning on of the central boiler is minimized.

I was thinking if the cyclus could always be configured from the whole hour, everything is synchronized. If the cyclus length for example is 30 minutes, the start times could be 13:00 and 13:30. Likewise if the cyclus length is 10 minutes, they could start at 13:00, 13:10, etc.

You always want to start the first cyclus after turning on the thermostat, because you don't want to wait for the next synchronized time. A solution for this could be to always use a short first cyclus that ends at the start of the next cyclus.

**Comment by jmcollin78** [comment]
Hello @tttopuz ,

There is no way to ensure all cycles are starting simultaneously. On the contrary, when over_switch are used it is mostly for electrical heater and you don't want to synchronize them to avoid pic of consumption.

In your case, I think that If you set all cycle to the same value, all heating phases will start at the same time. But, this will vary over time until the next HA restarts. So this is not a real solution.

**Reply by anthony0585** [reply]
not really if we change the temperature on a Vtherm it will start its calculation without the other.

**Comment by anthony0585** [comment]
Hello @jmcollin78 — I think this option would be useful. It would be especially helpful for a gas boiler. I’m in the same situation: if the devices are out of sync the boiler can end up cycling on and off too often.
If I set a 30-minute cycle for all my Vtherm units, I’d like all calculations to start together — when one Vtherm begins its calculation, it should trigger the others to start theirs. That way, during each 30-minute window, if the boiler starts we’ll have checked every Vtherm that needs it.

**Comment by jmcollin78** [comment]
No it is not a solution to your problem. If TPI calculate 10% of heating, with a cycle of 5 minutes, you will have 30 sec activated and 270 sec off. So it will always turn on and off.

The issue is that `over_switch` is for controling a switch which controls an electrical heater. For valve or central water boiler, you sould rather use `over_valve` or `over_climate` with direct valve regulation.

**Reply by anthony0585** [reply]
Thank you for your answer, but It's not what I mean, all my Vtherm have the same cycle duration 30minutes, but if there not sync there is too much start stop. Find what I mean there: 
<img width="692" height="184" alt="image" src="https://github.com/user-attachments/assets/49dce681-a2b4-4eb9-8518-586ed5f10155" />

or maybe like this:
 
<img width="685" height="182" alt="image" src="https://github.com/user-attachments/assets/f3166aab-92cc-4594-bc0b-ba3674c39131" />

In this way it's really better for the boiler. and it's exactly why I prefer using over_switch instead over_valve (because i don't want valve calling the boiler when it wants
I hope you will understand my mind. Actualy I found a way when a target temperature is changed I switch to off and then heat only on all Vtherm, but I think i swould be nice to have this function integrated.

**Comment by fricy** [comment]
Seconded, in gas boiler heated houses there's a lot of wasted energy heating up the pipes, the more cycles the bigger the waste. Imho the ideal solution is to minimize cycles at the expense of bigger temperature swings.

For eg, at the moment I have three TVR over_climate VTherms set up with a centralized switch, two needs to request heat for the boiler to start. It's really not ideal, because the rooms have different heat losses and the cycles are all over the place with lots of heat lost in the pipes that are outside coming in from the boiler room.

Perhaps one solution is to automatically open all the valves after a heating cycles stops and the boiler is off. Then any Vtherm can start the boiler, all of the rooms receive the heat, the valves close as they reach their own setpoint regulating the room, and the heating stops when the coldest room is regulated.

**Reply by anthony0585** [reply]
In your case, If I understood, I think you can simply uncheck "control the central boiler" (or check if there's an option to set the minimum number of VTherms required to trigger the central boiler — I'm not sure about this).
Then, create an automation that:

Starts your boiler when all your valves is requesting heat
Stops it when all valves are closed

In your automation, I think you should monitor the real valves, not the VTherms. That's because a VTherm over valve only adjusts the target temperature sent to the real valve — its purpose is to make your real valve operate according to your external temperature sensor, rather than its own built-in temperature.

**Comment by jmcollin78** [comment]
Hello I put it into a feature request but it will be hard to do and I don't like that. Vtherm has different cycle duration, different configuration (with command personnalized, ac_mode, TPI coefficients, ...). So grouping them make no sense if those properties are different. It will bring the complexity at a high level (and the level is already very high).

First I guess you was speeking about syncing multiple switch in the same VTherm. This is easy to do cause they all have the same properties. But syncing different Vtherm is another level of complexity.

Sorry.

**Reply by anthony0585** [reply]
I understand your concern, but let me reassure you.
To clarify my idea: this is not a permanent synchronization running all day long. It would be a "SYNC" option that forces a one-time resynchronization of cycles only when an action triggers a cycle restart on one of the VTherms (changing target temperature, modifying parameters, etc.).
The principle:

When an action forces a VTherm to restart its cycle, it also forces an immediate cycle restart for all other VTherms (or a configured group)
After that, each VTherm lives its life normally — no continuous sync mechanism
To maintain synchronization over time, the user must configure identical (or multiple) cycle durations

Concrete example:

V1, V2, V3 with 30-min cycles, currently offset (6:30 / 7:00 / 7:30)
At 6:50, I change V1's target temperature → this forces V1 to restart its cycle
With SYNC enabled → V2 and V3 are also forced to restart their cycle at 6:50
Next cycles for all: 7:20, 7:50, 8:20...

The sync only happens at that specific moment. It's not an automatic background process — it's purely triggered by actions that would normally restart a single VTherm's cycle. Are you ok with this ?

**Reply by jmcollin78** [reply]
With my words: each time a new cycle is started on one Vtherm, all VTherm will restart a cycle ?
Is that what you mean ?

Cycle are not restarted if already started but only the duration of the on_percent and off_percent is recalculted. The exception is when you do mode off and then mode heat.

This is definitively a bad idea sorry. I will no do that. Yo can try on your own if are able to code and see that things are not build as yo want.
